app/src/org/commcare/android/resource/installers/OfflineUserRestoreAndroidInstaller.java

```diff
@@ -0,0 +1,97 @@
+package org.commcare.android.resource.installers;
+
+import org.commcare.CommCareApplication;
+import org.commcare.android.database.app.models.UserKeyRecord;
+import org.commcare.models.database.user.DatabaseUserOpenHelper;
+import org.commcare.resources.model.Resource;
+import org.commcare.resources.model.ResourceInitializationException;
+import org.commcare.resources.model.UnresolvedResourceException;
+import org.commcare.suite.model.OfflineUserRestore;
+import org.commcare.utils.AndroidCommCarePlatform;
+import org.javarosa.core.reference.Reference;
+import org.javarosa.core.services.storage.EntityFilter;
+import org.javarosa.core.util.externalizable.DeserializationException;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @author Phillip Mates (pmates@dimagi.com)
+ * @author Aliza Stone (astone@dimagi.com)
+ */
+public class OfflineUserRestoreAndroidInstaller extends FileSystemInstaller {
+
+    @SuppressWarnings("unused")
+    public OfflineUserRestoreAndroidInstaller() {
+        // for externalization
+    }
+
+    public OfflineUserRestoreAndroidInstaller(String localDestination, String upgradeDestination) {
+        super(localDestination, upgradeDestination);
+    }
+
+    @Override
+    public boolean initialize(AndroidCommCarePlatform instance, boolean isUpgrade) throws ResourceInitializationException {
+        if (localLocation == null) {
+            throw new ResourceInitializationException("The user restore file location is null!");
```

Comment:
This should crash have crashed before now, right? the installer shouldn't be able to save itself without this flag being set, right? I feel like it's better to crash on save than it is to crash during runtime